src/cloud/components/onboarding/EngagementLink.tsx

```diff
@@ -0,0 +1,47 @@
+// Libraries
+import {FC, useEffect} from 'react'
+import {useLocation} from 'react-router-dom'
+import {connect, ConnectedProps} from 'react-redux'
+import {get} from 'lodash'
+
+// Utils
+import {getOrg} from 'src/organizations/selectors'
+
+// Types
+import {AppState} from 'src/types'
+
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = ReduxProps
+
+const EngagementLink: FC<Props> = ({org, me}) => {
+  const pathname = useLocation().pathname
+
+  useEffect(() => {
+    sendToUserPilot()
+  }, [pathname, org, me])
+
+  const sendToUserPilot = (): void => {
+    const userpilot = get(window, 'userpilot')
+    const host = window.location.hostname.split('.')
+
+    if (userpilot && org && me) {
+      userpilot.identify(me.name, {
+        email: me.name, // User Email address
+        orgID: org.id, // Organization ID
+        region: host[0], // Cloud provider region
+        provider: host[1], // Cloud provider
+      })
+    }
+  }
+
+  return null
+}
+
+const mstp = (state: AppState) => {
```

Comment:
there's a super cool hook called useSelector now that cleans this up a bit!
```
import {useSelector} from 'react-redux'

... inside your component ...
  const org = useSelector(getOrg)
  const me = useSelector((state: AppState) => state.me)
  ```
  
  and that's it! no props to worry about or wrapper functions to copy pasta

Reply:
Ooh, nice! I'll update.